x-pack/solutions/security/plugins/security_solution/server/endpoint/routes/suggestions/utils.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+/**
+ * Combines an index name with integration namespaces.
+ *
+ * @param indexName - The base index name (must end with ".*" or "-*")
+ * @param integrationNamespaces - Object containing integration names as keys and namespace arrays as values
+ * @param integrationName - The integration name to look up in the integrationNamespaces object (e.g., "endpoint")
+ * @returns Combined index names with namespaces, multiple indices separated by commas. Returns empty string if validation fails.
+ */
+
+export function combineIndexWithNamespaces(
```

Comment:
FYI: we already have utility to do this - see here:

https://github.com/elastic/kibana/blob/da25d13a2ac5be97216e202bf5d6bda6ff920b3b/x-pack/solutions/security/plugins/security_solution/common/endpoint/utils/index_name_utilities.ts#L20

Reply:
@paul-tavares 
It seems to me that this utility isn't compatible with the `logs-endpoint.events.*` pattern. With space awareness, we want to keep the `*` in the pattern and append the namespace after it, separated by a dash. For example, the expected output should be `logs-endpoint.events.*-default`.

Right now, the function produces `logs-endpoint.events.-default` for this pattern, which I believe isn't correct. We'll need to update the logic or switch to using our own parser. Please let me know what you think!

Reply:
There is this `.file` below the definition of FILE_EVENTS_INDEX_PATTERN - https://github.com/szwarckonrad/kibana/blob/osquery-space-awarness/x-pack/solutions/security/plugins/security_solution/common/endpoint/constants.ts#L20

Reply:
Ok. Thanks. But do you have actually examples of data that the endpoint streamed to ES that seems to use this pattern?

The reason I'm asking is that this does not look like something that should be happening - the endpoint sending data to a `namespace` that is not user defined (prepending a static string - in this case `file` to the `namespace` received from the fleet policy.